test/plugin/test_out_google_cloud.rb

```diff
@@ -70,6 +70,50 @@ def test_server_error
     assert_equal 1, exception_count
   end
 
+  def test_prometheus_metrics
+    setup_gce_metadata_stubs
+    [
+      # Single successful request.
+      [200, 1, 1, [1, 0, 1, 0]],
+      # Several successful requests.
+      [200, 2, 1, [2, 0, 2, 0]],
+      # Single successful request with several entries.
+      [200, 1, 2, [1, 0, 2, 0]],
+      # Single failed request that causes logs to be dropped.
+      [401, 1, 1, [0, 1, 0, 1]],
+      # Single failed request that escalates without logs being dropped.
+      [500, 1, 1, [0, 1, 0, 0]]
+    ].each do |code, request_count, entry_count, metric_values|
+      setup_prometheus
+      stub_request(:post, WRITE_LOG_ENTRIES_URI)
```

Comment:
We should be able to do this as part of `setup_logging_stubs`. Can you please add a TODO to refactor it later? Don't put a name -- we'll pick it up.